pkg/controller/common/helpers.go

```diff
@@ -1397,3 +1399,21 @@ func GetCAsFromConfigMap(cm *corev1.ConfigMap, key string) ([]byte, error) {
 	}
 	return nil, fmt.Errorf("%s not found in %s/%s", key, cm.Namespace, cm.Name)
 }
+
+// Determines if an on-cluster layering rebuild is required for the changes
+func RequiresRebuild(oldMC, newMC *mcfgv1.MachineConfig) bool {
+	return oldMC.Spec.OSImageURL != newMC.Spec.OSImageURL ||
+		oldMC.Spec.KernelType != newMC.Spec.KernelType ||
+		!unorderedEqual(oldMC.Spec.Extensions, newMC.Spec.Extensions) ||
```

Comment:
issue: To keep behavior consistent with the non-OCL path as well as keep the rest of the system in a consistent state, we should replace these `unorderedEqual()` checks with `reflect.DeepEqual()` instead. Here's why:

Although `rpm-ostree` ultimately doesn't care what order the extensions packages are in, order matters elsewhere. The `HashMachineConfigSpec()` function considers order whenever it produces a hashed MachineConfig name. This means we can end up in a situation where a new container-config gets created, but a new MachineOSBuild does not get created for it.

You can see an example of this here:
```console
$ echo '["ext1", "ext2"]' | md5sum
43409e8c44c08862bf5be1943e59e731  -

$ echo '["ext2", "ext1"]' | md5sum
4d0bbe46bc2e8e2cb4ac00fb63fbc550  -
```